src/test/java/com/puppycrawl/tools/checkstyle/internal/ImmutabilityTest.java

```diff
@@ -0,0 +1,433 @@
+            .importPackages("com.puppycrawl.tools.checkstyle.checks");
+
+        final ModulePropertyPredicate moduleProperties = new ModulePropertyPredicate();
+
+        final DescribedPredicate<JavaClass> immutableFields = new ImmutableFieldsPredicate();
+
+        final ArchCondition<JavaField> bePrivateExceptMessageKeys =
+            new ArchCondition<>("be private") {
+
+                @Override
+                public void check(JavaField item, ConditionEvents events) {
+                    if (!item.getName().startsWith("MSG")
+                        && !item.getModifiers().contains(JavaModifier.PRIVATE)) {
+                        final String message = String
+                            .format(Locale.ROOT, "Field <%s> should %s in %s",
+                                    item.getFullName(), getDescription(),
+                                    item.getSourceCodeLocation().toString());
+                        events.add(SimpleConditionEvent.violated(item, message));
+                    }
+                }
+            };
+
+        final ArchCondition<JavaClass> beAnnotatedOrSuppressed = new ArchCondition<>(
+            "be annotated or added to the suppression list") {
+            @Override
+            public void check(JavaClass item, ConditionEvents events) {
+                if (!SUPPRESSED_CLASSES.contains(item.getFullName())) {
+                    final String message = String
+                        .format(Locale.ROOT, "should %s.", getDescription());
+                    events.add(SimpleConditionEvent.violated(item, message));
+                }
+            }
+        };
+
+        final ArchCondition<JavaField> beSuppressed = new ArchCondition<>("be suppressed") {
+            @Override
+            public void check(JavaField item, ConditionEvents events) {
+                if (!SUPPRESSED_FIELDS_IN_MODULES.contains(item.getFullName())) {
+                    final String message = String
+                        .format(Locale.ROOT, "should %s.", getDescription());
+                    events.add(SimpleConditionEvent.violated(item, message));
+                }
+            }
+        };
+
+        final ArchRule fieldsInStatelessChecksAreImmutable = fields()
+            .that()
+            .areDeclaredInClassesThat()
+            .areAnnotatedWith(StatelessCheck.class)
+            .and(are(not(moduleProperties)))
+            .should(BE_IMMUTABLE)
+            .andShould()
+            .beFinal()
+            .andShould(bePrivateExceptMessageKeys)
+            .orShould(beSuppressed);
+
+        final ArchRule classesWithImmutableFieldsShouldBeStateless = classes()
+            .that(have(immutableFields))
+            .and()
+            .haveSimpleNameEndingWith("Check")
+            .and()
+            .doNotHaveModifier(JavaModifier.ABSTRACT)
+            .should()
+            .beAnnotatedWith(StatelessCheck.class)
+            .orShould(beAnnotatedOrSuppressed);
```

Comment:
Let's avoid `Or` in our predicates, and rely on build in `.orShould` to keep predicates as simple as possible.

Reply:
The variable name here was incorrect which caused the confusion, corrected now.